app/src/main/java/com/navapp/navigation/requests/PlaceAutocomplete.java

```diff
@@ -0,0 +1,25 @@
+package com.navapp.navigation.requests;
+
+import com.google.maps.GeoApiContext;
+import com.google.maps.PlaceAutocompleteRequest;
+import com.google.maps.PlacesApi;
+import com.google.maps.errors.ApiException;
+import com.google.maps.model.AutocompletePrediction;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+public class PlaceAutocomplete {
```

Comment:
Hmmm unused... What the point for this one?

Reply:
It will be used when the user will be searching for places, I added this now because it's from the google maps API

Reply:
I have a feeling the class is going to be changed a lot to fit working with UI. Just remember that it uses `await` when you use it in GUI, because it might be problematic.

Reply:
OK boomer, exceptions...

Reply:
Exceptions are nice and all, but have you considered that calling `await` from the GUI thread will just hang the UI, making it unresponsive.

Reply:
I see the issue now, we will probably add some sort a loading circle of death in the UI. But I honestly don't what's the best way to implement this. (mostly since I don't know how it will look in the UI side) So I prefer to leave it in this simple way and probably change later. (I don't think it would be that hard to do so)

Reply:
So... what should I do? you did not address this.

Reply:
> I see the issue now, we will probably add some sort a loading circle of death in the UI. But I honestly don't what's the best way to implement this. (mostly since I don't know how it will look in the UI side) So I prefer to leave it in this simple way and probably change later. (I don't think it would be that hard to do so)

I feel like you misunderstood, I am talking about this comment in particular. Can you reply to it ?

Reply:
You misunderstood
> GUI thread will just hang the UI, making it unresponsive

The UI will not work __AT ALL__. At least that's the case for desktop application, it's probably same is true for _Android_.

It's an `async` API, and should be used as such. That is, you don't wait for a response sequentially, but rather provide a _callback_, use:
```Java
PlacesApi.placeAutocomplete(context, address, sessionToken).setCallback(new PendingResult.Callback<AutocompletePrediction[]>() {
    @Override
    public void onResult(AutocompletePrediction[] result) {

    }

    @Override
    public void onFailure(Throwable e) {

    }
});
```

This also effectively changes the API of `PlaceAutocomplete` to an _async_ structure.

Reply:
didn't know it's a thing, I guess it's much simpler then I thought

Reply:
done